analytics/analytics.go

```diff
@@ -5,91 +5,121 @@ import (
+	Runtime       time.Duration   `json:"run_time"`
+	Platform      string          `json:"platform"`
+	BuildSlug     string          `json:"build_slug"`
+	StartTime     time.Time       `json:"start_time"`
+	CLIVersion    string          `json:"cli_version"`
+	StepAnalytics []StepAnalytics `json:"step_analytics"`
 }
 
-// AnonymizedUsageGroupModel ...
-type AnonymizedUsageGroupModel struct {
-	Steps []AnonymizedUsageModel `json:"steps"`
+// StepAnalytics ...
+type StepAnalytics struct {
+	StepID    string        `json:"step_id"`
+	Status    string        `json:"status"`
+	Runtime   time.Duration `json:"run_time"`
+	StartTime time.Time     `json:"start_time"`
 }
 
 //=======================================
 // Main
 //=======================================
 
-// SendAnonymizedAnalytics ...
-func SendAnonymizedAnalytics(buildRunResults models.BuildRunResultsModel) error {
-	stepRunResults := buildRunResults.SuccessSteps
-	stepRunResults = append(stepRunResults, buildRunResults.FailedSteps...)
-	stepRunResults = append(stepRunResults, buildRunResults.FailedSkippableSteps...)
-	stepRunResults = append(stepRunResults, buildRunResults.SkippedSteps...)
+func buildStatus(buildFailed bool) string {
+	return map[bool]string{false: "successful", true: "failed"}[buildFailed]
+}
 
-	anonymizedUsageGroup := AnonymizedUsageGroupModel{}
-	for _, stepRunResult := range stepRunResults {
-		anonymizedUsageData := AnonymizedUsageModel{
-			ID:      stepRunResult.StepInfo.ID,
-			Version: stepRunResult.StepInfo.Version,
-			RunTime: stepRunResult.RunTime.Seconds(),
-			Error:   stepRunResult.Status != 0,
-			CI:      configs.IsCIMode,
-		}
+func stepStatus(i int) string {
+	if status, ok := map[int]string{
+		models.StepRunStatusCodeFailed:           "failed",
+		models.StepRunStatusCodeSuccess:          "success",
+		models.StepRunStatusCodeSkipped:          "skipped",
+		models.StepRunStatusCodeFailedSkippable:  "failed_skippable",
+		models.StepRunStatusCodeSkippedWithRunIf: "skipped_with_runif",
+	}[i]; ok {
+		return status
+	}
+	return "unknown"
+}
 
-		anonymizedUsageGroup.Steps = append(anonymizedUsageGroup.Steps, anonymizedUsageData)
+// SendAnonymizedAnalytics ...
+func SendAnonymizedAnalytics(buildRunResults models.BuildRunResultsModel) error {
+	var (
+		runtime       time.Duration
+		stepAnalytics []StepAnalytics
+	)
+	for _, stepResult := range buildRunResults.OrderedResults() {
+		stepAnalytics, runtime = append(stepAnalytics, StepAnalytics{
+			StepID:    stepResult.StepInfo.ID,
+			Status:    stepStatus(stepResult.Status),
+			Runtime:   stepResult.RunTime,
+			StartTime: stepResult.StartTime,
+		}), runtime+stepResult.RunTime
 	}
 
-	data, err := json.Marshal(anonymizedUsageGroup)
-	if err != nil {
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(BuildAnalytics{
+		Runtime:    runtime,
+		StartTime:  buildRunResults.StartTime,
+		Platform:   buildRunResults.ProjectType,
+		StackID:    os.Getenv(stackIDEnvKey),
```

Comment:
Some format/lenght check would be nice, to prevent sending bad data if the user has these set to a wrong value.

Reply:
this will be set by the website, and need to send this data with any data in it

Reply:
could be set manually when running locally, agree that not too propable